File: app/src/main/java/com/radiusnetworks/powerconnecttester/PowerConnectionReceiver.java
package com.radiusnetworks.powerconnecttester;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * Created by dyoung on 1/21/15.
 */
public class PowerConnectionReceiver extends BroadcastReceiver {
    public static final String TAG = "PowerConnectionReceiver";

    public void onReceive(Context context , Intent intent) {
        String action = intent.getAction();

        String actionName = null;

        if(action.equals(Intent.ACTION_POWER_CONNECTED)) {
            actionName = "ACTION_POWER_CONNECTED";
        }
        else if(action.equals(Intent.ACTION_POWER_DISCONNECTED)) {
            actionName = "ACTION_POWER_DISCONNECTED";
        }
        else if(action.equals(Intent.ACTION_BOOT_COMPLETED)) {
            actionName = "ACTION_BOOT_COMPLETED";
        }
        else  {
            actionName = "OTHER: "+action;
        }
        Intent newIntent = new Intent(context, MainActivity.class);
        newIntent.putExtra("action", actionName);
        newIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        Log.d(TAG, "Sending intent: "+newIntent+" for broadcast intent: "+intent);
        context.getApplicationContext().startActivity(newIntent);
    }
}

File: power_connect_tester/PowerConnectionReceiver.cs
using System;
using Android.App;
using Android.Content;
using Android.Util;

namespace power_connect_tester
{
    [BroadcastReceiver]
    [IntentFilter(new[] {
        Intent.ActionBootCompleted,
        Intent.ActionPowerConnected,
        Intent.ActionPowerDisconnected
    })]
    public class PowerConnectionReceiver : BroadcastReceiver
    {
        public const string TAG = "PowerConnectionReceiver";

        public override void OnReceive(Context context, Intent intent)
        {
            String action = intent.Action;

            String actionName = null;

            if (action.Equals(Intent.ActionPowerConnected))
            {
                actionName = "ACTION_POWER_CONNECTED";
            }
            else if (action.Equals(Intent.ActionPowerDisconnected))
            {
                actionName = "ACTION_POWER_DISCONNECTED";
            }
            else if (action.Equals(Intent.ActionBootCompleted))
            {
                actionName = "ACTION_BOOT_COMPLETED";
            }
            else
            {
                actionName = "OTHER: " + action;
            }

            Intent newIntent = new Intent(context, typeof(MainActivity));
            newIntent.PutExtra("action", actionName);
            newIntent.SetFlags(ActivityFlags.NewTask);

            Log.Debug(TAG, "Sending intent: "+ newIntent + " for broadcast intent: " + intent);
            context.ApplicationContext.StartActivity(newIntent);
        }
    }
}

